apps/server/src/modules/learnroom/service/course-sync.service.spec.ts

```diff
@@ -395,11 +449,69 @@ describe(CourseSyncService.name, () => {
 						name: course.name,
 						startDate: newGroup.validPeriod?.from,
 						untilDate: newGroup.validPeriod?.until,
-						studentIds: [],
+						studentIds: [studentUserId],
 						teacherIds: [teacherUserId],
 						syncedWithGroup: course.syncedWithGroup,
 						classIds: [],
 						groupIds: [],
+						excludeFromSync: [SyncAttribute.TEACHERS],
+						substitutionTeacherIds: [substituteTeacherId],
+					}),
+				]);
+			});
+		});
+
+		describe('when there teachers are no teachers synced from group', () => {
+			const setup = () => {
+				const substituteTeacherId = new ObjectId().toHexString();
+				const studentUserId = new ObjectId().toHexString();
+				const teacherUserId = new ObjectId().toHexString();
+				const studentRoleId: string = new ObjectId().toHexString();
+				const studentRole: RoleDto = roleDtoFactory.build({ id: studentRoleId });
+				const teacherRole: RoleDto = roleDtoFactory.build();
+				const newGroup: Group = groupFactory.build({
+					users: [
+						new GroupUser({
+							userId: studentUserId,
+							roleId: studentRoleId,
+						}),
+					],
+				});
+
+				const course: Course = courseFactory.build({
+					syncedWithGroup: newGroup.id,
+					substitutionTeacherIds: [substituteTeacherId],
+					excludeFromSync: [],
+				});
+				courseRepo.findBySyncedGroup.mockResolvedValueOnce([new Course(course.getProps())]);
+				roleService.findByName.mockResolvedValueOnce(studentRole);
+				roleService.findByName.mockResolvedValueOnce(teacherRole);
+
+				return {
+					course,
+					newGroup,
+					teacherUserId,
+					substituteTeacherId,
+					studentUserId,
+				};
+			};
+
+			it('should not sync group students', async () => {
+				const { course, newGroup, substituteTeacherId } = setup();
+
+				await service.synchronizeCourseWithGroup(newGroup, newGroup);
+				expect(courseRepo.saveAll).toHaveBeenCalledWith<[Course[]]>([
+					new Course({
+						...course.getProps(),
+						name: course.name,
+						startDate: newGroup.validPeriod?.from,
+						untilDate: newGroup.validPeriod?.until,
+						studentIds: [],
+						teacherIds: [],
```

Comment:
With this setup your course doesn't have a teacher to begin with, which shouldn't be the case... The test before had a teacher in the course and the saveAll function was expected to be called with this teacher.